cmd/zq/README.md

```diff
@@ -0,0 +1,102 @@
+  format by default. For all `zq` options, see the built-in help by running:
+
+```
+zq help
+```
+
+## Examples
+
+Here are a few examples using a small Zeek formatted log file, `conn.log
+`, located in this directory. See the
+[zq-sample-data repo](https://github.com/brimsec/zq-sample-data) for more test
+data, which is used in the examples in the
+[query language documentation](../../zql/docs/README.md).
+
+To cut the columns of a Zeek "conn" log like `zeek-cut`, and output to the
+ terminal run:
+
+```
+zq -t "* | cut ts,id.orig_h,id.orig_p" conn.log
+```
+
+The `-t` tells `zq` to use [TZNG](../../zng/docs/spec.md#4-zng-text-format-tzng)
+for its output format. The "`*`" 
+tells `zq` to match every line, which is sent to the `cut` processor
+using the UNIX-like pipe syntax.
+
+When looking over everything like this, you can omit the search pattern
+as a shorthand and simply type:
+```
+zq -t "cut ts,id.orig_h,id.orig_p" conn.log
+```
+
+The default output is the binary ZNG format.  If you want just the tab-separated
+ lines like `zeek-cut`, you can specify text output:
+```
+zq -f text "cut ts,id.orig_h,id.orig_p" conn.log
+```
+If you want the old-style Zeek [ASCII TSV](https://docs.zeek.org/en/stable/examples/logs/)
+log format, run the command with the `-f` flag specifying `zeek` for the output
+format:
+```
+zq -f zeek "cut ts,id.orig_h,id.orig_p" conn.log
+```
+You can use an aggregate function to summarize data over one or
+more fields, e.g., summing field values, counting, or computing an average.
+```
+zq -t "sum(orig_bytes)" conn.log
+zq -t "orig_bytes > 10000 | count()" conn.log
+zq -t "avg(orig_bytes)" conn.log
+```
+
+The [ZNG specification](../../zng/docs/spec.md) describes the significance of
+ the
+`_path` field.  By leveraging this, diverse Zeek logs can be combined into a single
+file.
+```
+zq *.log > all.zng
+```
+
+### Comparisons
+
+Revisiting the `cut` example shown above:
+
+```
+zq -f text "cut ts,id.orig_h,id.orig_p" conn.log
+```
+
+This is functionally equivalent to the `zeek-cut` command-line:
+
+```
+zeek-cut ts id.orig_h id.orig_p < conn.log
+```
+
+If your Zeek events are stored as JSON and you are accustomed to querying with `jq`,
+the equivalent would be:
+
+```
+jq -c '. | { ts, "id.orig_h", "id.orig_p" }' conn.ndjson
+```
+
+Comparisons of other simple operations and their relative performance are described
+at the [performance](../../performance/README.md) page.
+
+
+## Formats
+
+| Format | Read | Auto-Detect | Write | Description |
+|--------|------|-------------|-------|-------------|
+| zng | yes | yes | yes | [ZNG specification](../../zng/docs/spec.md) |
+| tzng | yes | yes | yes | [TZNG specification](../../zng/docs/spec.md#4-zng-text-format-tzng) |
+| ndjson | yes | yes | yes | Newline delimited JSON records |
+| zeek  | yes | yes | yes | [Zeek compatible](https://docs.zeek.org/en/stable/examples/logs/) tab separated values |
+| zjson | yes | yes | yes | Zeek JSON |
```

Comment:
While many readers may not linger on this, strictly speaking, I think it would be inaccurate to call this "Zeek JSON", since users coming from the Zeek world are more likely to see this as a reference to [Zeek's regular ASCII logger outputting JSON](https://docs.zeek.org/en/current/frameworks/logging.html#ascii-writer) or the use of the [json-streaming-logs](https://github.com/corelight/json-streaming-logs) package. I don't really feel like I'm well-suited to say what we should say instead since my understanding is that this zjson format is something that was effectively created for internal developer use, and I'm not a developer. My suspicion is that it would more accurately be called "ZNG over JSON" and link to https://github.com/brimsec/zq/blob/zq-readme-update/zng/docs/zng-over-json.md (which I believe is a woefully out-of-date doc, alas).